### DISCUSSION TITLE
Welcome to ss3roms Discussions!

### DISCUSSION THREAD

**Discussion by kellijohnson-NOAA** [pacific-hake/ss3roms#2] [Announcements]
## SS3 run comparison

I did an initial check to make sure the code I wrote was working when adding a ROMS variable as an index of age0 fish by seeing if the files would run. Then, I compared the MLE output to a run with the Age1 index only. Recruitment estimates are different
![compare11_recdevs](https://user-images.githubusercontent.com/4108564/147635591-fb8a151e-3058-4ec0-8e0a-3041bef32d0c.png)
.

@aaronmberger-nwfsc @okenk @kristinmarshall-NOAA do you have any intuition on the reason why almost all estimates of recruitment deviations that are different between the two model runs are more similar to the median recruitment for the model with the upwelling index for age-0 fish (red) compared to a model with just the age-1 index (blue)?

**Comment by kristinmarshall-NOAA** [comment]
I've been thinking about this but don't have a good answer.  Here are some things I've considered:

- Cathleens fitted model for the recruitment deviations under-predicts the extremes (as did Nick and Melissa's for other species), but that shouldn't come into play because you are just using upwelling as an index of age-0.
-You see the effect most in the low recruitment years in the 1980s, 1990s, and 2000s. In Cathleen's analysis, strong upwelling had a negative relationship with recruitment, so we could see whether those years were strong upwelling years.
-DId you add upwelling as an index on log-recruits or total recruits?  That might also have something to do with it?

**Comment by okenk** [comment]
If strong (i.e., positive) upwelling had a negative relationship with recruitment, should we first be multiplying it by -1 to use it as a recruitment index? That could explain this pattern.

**Comment by okenk** [comment]
I switched to the eddy kinetic energy since that seems to be the strongest driver. I tried both the raw values as an age-0 index and -x + min(x) (SS got angry when I tried just -x, I think it was the negative numbers), since EKE is also negatively associated with recruitment. The rec devs were different for the two, but in both cases shrunk towards zero which was suspicious. I think the log thing could be an issue. There is a way to specify that the units of the index are in exp(recruitment).

**Comment by okenk** [comment]
Hmm, I am about to log off for the weekend, but am not having much success and thought I would document it here in case others have ideas. I have tried:
1. Specifying the index is for exp(rec dev) and using EKE
2. Specifying the index is for exp(rec dev) and using exp(EKE)
3. (1) and (2) but using -EKE + max(EKE) + 0.01 (i.e., inverting the signal and then offsetting to keep values positive)
All essentially lead to rec devs shrinking towards zero as in Kelli's plot.

I also tried increasing the CV for the index. The black dots in the index fits (is this the modeled value for the index?) become basically constant over time. Is that expected?

I also peeked into the sablefish assessment. I was a bit confused by the data file because at the top the units for the environmental driver are listed as 2 (F, can't be used for surveys?), and farther down they are listed as 31 (exp(rec devs), what I have been using). 

One thought I had is the signal looks fairly weak based on a scatter plot of EKE vs rec devs:
![image](https://user-images.githubusercontent.com/205013/149600453-dd6d8d58-5b2e-4f0f-b201-e0820bdf285f.png)
Is it possible the weak signal is causing the issue? @kristinmarshall-NOAA, I wonder what would happen if we use the predictions from the best-fitting model, instead of a single driver?

@kellijohnson-NOAA, @aaronmberger-nwfsc, @kristinmarshall-NOAA, it may be useful to have a conversation next week sometime.

**Comment by kellijohnson-NOAA** [comment]
I tried fitting the upwelling index as it was given b/c that is the example in add_fleet and q was estimated on the upper bound of 1.2. Exploring more now.

**Comment by kristinmarshall-NOAA** [comment]
Any more luck with this? 

Do you want me to send the predictions from Cathleen's best-fit model to try out?  

I was also thinking that the [bifurcation index](https://github.com/michaelmalick/bifurcation-index.git), while not from ROMS, is updated to 2020, thanks to Mike Malick and was probably the most robust predictor from Cathleen's analysis.  That might be another thing just to try because it could be used for a longer time period (1980-2020).

**Comment by okenk** [comment]
Not much. But I realized that the best way to test if the problem is
uninformative drivers is to just use the rec devs from the base model as
the drivers in the environmentally linked model. (Not that this makes any
sense scientifically, just as a diagnostic test.)

On Fri, Jan 21, 2022 at 1:11 PM Kristin Marshall NWFSC <
***@***.***> wrote:

> Any more luck with is?
>
> Do you want me to send the predictions from Cathleen's best-fit model to
> try out?
>
> I was also thinking that the bifurcation index
> <https://github.com/michaelmalick/bifurcation-index.git>, while not from
> ROMS, is updated to 2020, thanks to Mike Malick and was probably the most
> robust predictor from Cathleen's analysis. That might be another thing just
> to try because it could be used for a longer time period (1980-2020).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pacific-hake/ss3roms/discussions/2#discussioncomment-2018910>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABSBVMBJ5VMDQTO5BTU5FTUXHDY3ANCNFSM5K5NANGQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

*Kiva Oken, PhD (she/her <https://www.mypronouns.org/what-and-why>)*


*Research Mathematical StatisticianNOAA FisheriesU.S. Department of
Commerce*
(564) 225-2297
***@***.*** ***@***.***>
www.fisheries.noaa.gov
